src/Search.php

```diff
@@ -4501,6 +4501,15 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
                 $nott = !$nott;
                //negated, use contains case
             case "contains":
+                if ($searchopt[$ID]["datatype"] === 'decimal') {
+                    $matches = [];
+                    preg_match('/^(\d+.?\d?)/', $val, $matches);
+                    $val = $matches[1];
+                    if (!str_contains($val, '.')) {
+                        $val .= '.';
+                    }
```

Comment:
I guess a regexp is not really needed:
```suggestion
                    if (strpos($val, '.') === false) {
                        $val .= '.';
                    }
```

Reply:
Regex was just to ensure we get a clean number/decimal. If there are any spaces, zero-width characters (I've seen weird things when copying from some apps), wildcard characters if they are used to the old behavior, etc.